kernel/src/actions/mod.rs

```diff
@@ -589,6 +592,19 @@ impl Sidecar {
     }
 }
 
+/// The CheckpointMetadata action describes details about a checkpoint following the V2 specification.
+///
+/// [More info]: https://github.com/delta-io/delta/blob/master/PROTOCOL.md#checkpoint-metadata
+#[derive(Schema, Debug, PartialEq)]
```

Comment:
probably add the set we generally have for all actions
```suggestion
#[derive(Debug, Clone, PartialEq, Eq, Schema)]
```